gitlab/client.py

```diff
@@ -1039,10 +1039,10 @@ def next_page(self) -> Optional[int]:
         return int(self._next_page) if self._next_page else None
 
     @property
-    def per_page(self) -> int:
+    def per_page(self) -> Optional[int]:
         """The number of items per page."""
-        if TYPE_CHECKING:
-            assert self._per_page is not None
+        if self._per_page is None:
+            return None
```

Comment:
We're doing this slightly differently just above for `prev_page` and `next_page` to also handle empty headers, not sure how likely that is, just a note.

Reply:
Above is the same but just a single line. I will update this to match.